src/main/java/mate/academy/util/HashUtil.java

```diff
@@ -0,0 +1,32 @@
+package mate.academy.util;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+public class HashUtil {
+    private static final String CRYPTO_ALGORITHM = "SHA-512";
+    private static final int DEFAULT_SALT_ARRAY_SIZE = 16;
+
+    public static byte[] getSalt() {
+        SecureRandom secureRandom = new SecureRandom();
+        byte[] salt = new byte[DEFAULT_SALT_ARRAY_SIZE];
+        secureRandom.nextBytes(salt);
+        return salt;
+    }
+
+    public static String hashPassword(String password, byte[] salt) {
+        StringBuilder hashedPassword = new StringBuilder();
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance(CRYPTO_ALGORITHM);
+            messageDigest.update(salt);
+            byte[] digest = messageDigest.digest(salt);
+            for (byte b : digest) {
+                hashedPassword.append(String.format("%02x", b));
+            }
+        } catch (NoSuchAlgorithmException e) {
+            throw new IllegalStateException("Could not create hash using SHA-512 algorithm");
```

Comment:
The exception message 'Could not create hash using SHA-512 algorithm' is misleading. This exception is thrown when the specified algorithm is not available, not when there's a problem creating the hash. Consider changing the message to 'SHA-512 algorithm not available.'